app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/WatchBroadcast/WatchBroadcastService.java

```diff
@@ -0,0 +1,535 @@
+            bundle.putString("status.insulin", JoH.qs(statsResult.getTotal_insulin(), 2));
+            bundle.putString("status.royce_ratio", JoH.qs(statsResult.getRatio(), 2));
+            bundle.putString("status.capture_percentage", statsResult.getCapturePercentage(false));
+            bundle.putString("status.capture_realtime_capture_percentage", statsResult.getRealtimeCapturePercentage(false));
+            String accuracyString = "";
+            final long accuracy_period = DAY_IN_MS * 3;
+            final String accuracy_report = Accuracy.evaluateAccuracy(accuracy_period);
+            if ((accuracy_report != null) && (accuracy_report.length() > 0)) {
+                accuracyString = accuracy_report;
+            } else {
+                final String accuracy = BloodTest.evaluateAccuracy(accuracy_period);
+                accuracyString = (accuracy != null) ? " " + accuracy : "";
+            }
+            bundle.putString("status.accuracy", accuracyString);
+            bundle.putString("status.steps", String.valueOf(statsResult.getTotal_steps()));
+            
+            watchBroadcast.setStatCache(bundle, statHours);
+        }
+        return bundle;
+    }
+
+    public Bundle prepareBgBundle(WatchBroadcast watchBroadcast) {
+        if (watchBroadcast == null) return null;
+        WatchSettings settings = watchBroadcast.getSettings();
+        if (settings == null) return null;
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(xdrip.getAppContext());
+
+        Bundle bundle = new Bundle();
+        bundle.putBoolean("doMgdl", (prefs.getString("units", "mgdl").equals("mgdl")));
+        bundle.putInt("phoneBattery", PowerStateReceiver.getBatteryLevel(xdrip.getAppContext()));
+
+        BestGlucose.DisplayGlucose dg = BestGlucose.getDisplayGlucose();
+        BgReading bgReading = BgReading.last();
+        if (dg != null || bgReading != null) {
+            String deltaName;
+            double bgValue;
+            boolean isBgHigh = false;
+            boolean isBgLow = false;
+            boolean isStale;
+            long timeStamp;
+            String plugin = "";
+            double deltaValue = 0;
+            if (dg != null) {
+                deltaName = dg.delta_name;
+                //fill bg
+                bgValue = dg.mgdl;
+                isStale = dg.isStale();
+                isBgHigh = dg.isHigh();
+                isBgLow = dg.isLow();
+                timeStamp = dg.timestamp;
+                plugin = dg.plugin_name;
+                deltaValue = dg.delta_mgdl;
+            } else {
+                deltaName = bgReading.getDg_deltaName();
+                bgValue = bgReading.getDg_mgdl();
+                isStale = bgReading.isStale();
+                timeStamp = bgReading.getEpochTimestamp();
+            }
+            bundle.putString("bg.deltaName", deltaName);
+            bundle.putDouble("bg.valueMgdl", bgValue);
```

Comment:
yes, it's like a raw value, the conversion would be on the receiver side if needed . But maybe you right, it would be better to send value only in xdrip value format. I don't this someone would need to display bg value in a different format than configured in xdrip.